### PR TITLE
Handle mirrors that don't return a correct X-PYPI-LAST-SERIAL header

### PR DESCRIPTION
**Pull request by elesel**
I'm using a PyPI mirror provides a "None" value for the X-PYPI-LAST-SERIAL in its responses, and I thought others might run into the same issue.

### CODE REVIEW

File: server/devpi_server/extpypi.py [hunk 1]
```diff
@@ -416,7 +416,10 @@ def get_simplelinks_perstage(self, project):
         # Returning a 200 with "no such project" was originally meant to
         # provide earlier versions of easy_install/pip to request the full
         # simple page.
-        serial = int(response.headers.get(str("X-PYPI-LAST-SERIAL"), "-1"))
+        try:
+            serial = int(response.headers.get(str("X-PYPI-LAST-SERIAL"), "-1"))
```
Comment (fschulze) [inline server/devpi_server/extpypi.py]: I'd remove the "-1". That way the code path is the same way for any missing or invalid header combination.

**Comment by fschulze** [comment]
Thanks for this! For the changelog entry I'd classify this as a bugfix.

**Comment by elesel** [comment]
I believe I've done what you asked. Please let me know if you need anything else.

**Comment by fschulze** [comment]
I've expanded a test and made some minor changes. Will merge after test run.

**Comment by fschulze** [comment]
The test failures come from Pyramid 1.10a1 and will be fixed with the next release.

**Comment by elesel** [comment]
Thanks!

**Pull request merged**